core/chains/evm/client/chain_client.go

```diff
@@ -269,3 +270,12 @@ func (c *chainClient) TransactionReceipt(ctx context.Context, txHash common.Hash
 func (c *chainClient) LatestFinalizedBlock(ctx context.Context) (*evmtypes.Head, error) {
 	return c.multiNode.LatestFinalizedBlock(ctx)
 }
+
+func (c *chainClient) CheckTxValidity(ctx context.Context, from common.Address, to common.Address, data []byte) error {
```

Comment:
How should the product understand if the received error is OOC type?

Reply:
Since we standardize the error we send back, they could use [ErrOutOfCounters](https://github.com/smartcontractkit/chainlink/pull/12503/files#diff-ed85d15901e7cfbfdfa625b140d903c8ec415cf09c24373330e695b874b3053aR14) to determine if the error is an OOC error. I think to make it simpler for them though we'd maybe need to return a second output either another error or a flag. But I took this approach in efforts to keep the return a single field.